builtins/amp-img.js

```diff
@@ -64,17 +64,12 @@ export class AmpImg extends BaseElement {
   }
 
   /** @override */
-  prerenderAllowed() {
-    return true;
-  }
-
-  /** @override */
   isRelayoutNeeded() {
     return true;
   }
 
   /** @override */
-  layoutCallback() {
+  prerenderCallback() {
```

Comment:
True for the `amp-img` case. In this setup I don't see a problem of us doing a lower quality in prerender and then upgrading it to full quality on layout. Where do you see the problem?

As for the second paragraph. The runtime maintains 2 separate states for layout and prerender. I am not sure if the "layout should be almost instantaneous" refers to only the image case, but assuming so, then yes this would be true for cached image resources if they weren't unloaded on unlayout. 

`amp-img` has been using prerender mechanism through prerenderAllowed when it is basically layout, I just wanted to upgrade the usage to the new flow without disrupting the logic too much and digging into it too deep.